src/libstd/sys/common/remutex.rs

```diff
@@ -233,9 +233,8 @@ mod tests {
             *lock.borrow_mut() = 1;
             let lock2 = mc.lock().unwrap();
             *lock.borrow_mut() = 2;
-            let answer = Answer(lock2);
+            Answer(lock2);
```

Comment:
I think this may have slightly changed the semantics here as `Answer` lives over the call to `panic!`,  but changing the declaration to `let _answer = ...` and removing the `drop` should have the same semantics.